### DISCUSSION TITLE
中文字体显示

### DISCUSSION THREAD

**Discussion by YSW33** [AvaloniaUI/Avalonia#12468] [General]
项目里面Desktop中文字体显示正常，切换到Browser全是方块。

**Comment by timunie** [comment]
@YSW33 please try to ask your questions in english. Thx 🙏 

If I understand you correctly, you cannot show chinese chars in browser. Most likely the font is not avialable and needs to be added to your project as AvaloniaResource.

**Reply by YSW33** [reply]
Thank you。

**Comment by kyuranger** [comment]
It seems the same problem while running in Linux with Chinese character. The solution is embedding your own font (Here I am using the Noto Han Sans: https://github.com/adobe-fonts/source-han-sans/blob/master/README-CN.md ) in your project and set its build action to AvaloniaResources, just like this:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/2ec93fc7-032b-473f-b3cb-ea29e81cf95f)

Then in your MainWindow or MainView reference it like this:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/a01c1416-bf45-4c77-8d5b-431e315869c9)

`FontFamily="avares://Your Project Name/Fonts/#Source Han Sans"`

Finally, it may work now. Hope I can help you.

**Reply by YSW33** [reply]
Thank you, I'll give it a try。

**Reply by robloo** [reply]
Just a little unrelated comment on project structure (feel free to ignore). Usually fonts, images, etc. would go under an `Assets` folder or in older projects a `Resources` folder (like shown below).

**Comment by Coloryr** [comment]
我用的小米字体（思源黑体的太大了
![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/352a6a19-68b5-4af0-9b93-ef2d71460d7a)
```
public const string Font = "resm:ColorMC.Launcher.Resources.MiSans-Normal.ttf?assembly=ColorMC.Launcher#MiSans";
public static AppBuilder BuildAvaloniaApp()
{
return AppBuilder.Configure<App>()
    .With(new FontManagerOptions
    {
        DefaultFamilyName = Font,
    })
    .With(opt)
    .With(opt1)
    .LogToTrace()
    .UsePlatformDetect();
}
```

**Reply by YSW33** [reply]
按照你这个设置完，工程报错了Could not create glyphTypeface，很奇怪。

**Reply by YSW33** [reply]
是我自己犯蠢，字体名称打错了一个字符……
THANK YOU!!!

**Reply by YSW33** [reply]
问题解决了。

**Comment by Flithor** [comment]
Notice: Font file may have copyright, and may not free for commercial software. Embedding it in software without liscense may face legal risks. Know the all things clearly of the font you want to use.

**Reply by YSW33** [reply]
Thank you
